src/common/readfield_fi.f90

```diff
@@ -180,10 +180,7 @@ subroutine readfield_fi(istep, nhleft, itimei, ihr1, ihr2, itimefi, ierror)
     end if
 
 ! time between two inputs
-! open the correct file, if required
-    if (file_name /= filef(iavail(ntav2)%fileNo)) then
-      call check(fio%close (), "close fio")
-    end if
```

Comment:
We must have had a use-after-free here (fio should have been `save`). Well spotted!

Reply:
It was rather a bad copy from the readfield_nc.f90, where we keep the filehandle open and stored in snap.F90.
For now I reopen the meteo-file each time we read meteo with fimex. If we see there's a performance bottleneck we can add a `save`. I don't really like keeping files open for a long time.